youtube_dl/extractor/tube8.py

```diff
@@ -1,86 +1,260 @@
 from __future__ import unicode_literals
 
+import itertools
 import re
+from time import sleep
 
+from .common import InfoExtractor
 from ..utils import (
+    traverse_obj,
+    T,
+    url_or_none,
+    parse_iso8601,
+    get_element_by_class,
+    get_element_by_id,
     int_or_none,
-    str_to_int,
+    parse_qs,
+    urljoin,
 )
-from .keezmovies import KeezMoviesIE
 
 
-class Tube8IE(KeezMoviesIE):
-    _VALID_URL = r'https?://(?:www\.)?tube8\.com/(?:[^/]+/)+(?P<display_id>[^/]+)/(?P<id>\d+)'
+class Tube8IE(InfoExtractor):
+    _VALID_URL = r'https?:\/\/(?:www\.)?tube8\.com\/+porn-video+\/(?P<id>\d+)'
     _TESTS = [{
-        'url': 'http://www.tube8.com/teen/kasia-music-video/229795/',
-        'md5': '65e20c48e6abff62ed0c3965fff13a39',
+        'url': 'https://www.tube8.com/porn-video/189530841/',
+        'md5': '532408f59e89a32027d873af6289c85a',
         'info_dict': {
-            'id': '229795',
-            'display_id': 'kasia-music-video',
+            'id': '189530841',
             'ext': 'mp4',
-            'description': 'hot teen Kasia grinding',
-            'uploader': 'unknown',
-            'title': 'Kasia music video',
-            'age_limit': 18,
-            'duration': 230,
-            'categories': ['Teen'],
-            'tags': ['dancing'],
+            'title': 'Found dildo. She let it cum in her tight ass to keep the secret',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader': 'MaryKrylova',
+            'timestamp': 1718961736,
+            'upload_date': '20240621',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_valid_url(url).group('id')
+        webpage = self._download_webpage(url, video_id)
+
+        playervars = self._search_json(
+            r'\bplayervars\s*:', webpage, 'playervars', video_id)
+
+        extra_info = self._search_json(
+            r'application/ld\+json[\"\']?>\s?', webpage, 'extra_info', video_id)
+
+        uploader = traverse_obj(extra_info, (
+            '@graph', lambda _, v: v.get('author'), 'author'))[0]
+
+        thumbnail = traverse_obj(extra_info, (
+            '@graph', lambda _, v: v.get('thumbnail'), 'thumbnail'))[0]
+
+        timestamp = parse_iso8601(traverse_obj(extra_info, (
+            '@graph', lambda _, v: v.get('datePublished'), 'datePublished'))[0])
+
+        # Borrowed from youporn extractor
+        def get_fmt(x):
+            v_url = url_or_none(x.get('videoUrl'))
+            if v_url:
+                x['videoUrl'] = v_url
+                return (x['format'], x)
+
+        defs_by_format = dict(traverse_obj(playervars, (
+            'mediaDefinitions', lambda _, v: v.get('format'), T(get_fmt))))
+
+        title = traverse_obj(playervars, 'video_title')
+        if not thumbnail:
+            thumbnail = traverse_obj(playervars, 'image_url')
```

Comment:
With your suggestion, I got a KeyError while running the test. So it would be better to check if the thumbnail key is present with the following:

```suggestion
        if 'thumbnail' not in info.keys():
            info['thumbnail'] = traverse_obj(playervars, ('image_url', T(url_or_none)))
```

Reply:
Or even `if not info.get('thumbnail'):`, which is what I meant to write.